icicle-vm/src/lib.rs

```diff
@@ -291,13 +291,20 @@ impl Vm {
         }
 
         match self.lift(pc) {
-            Some(group) => {
+            Ok(group) => {
                 self.cpu.block_id = group.blocks.0 as u64;
                 self.cpu.block_offset = 0;
                 VmExit::Running
             }
-            None => {
-                self.cpu.exception = cpu::Exception::new(ExceptionCode::InvalidInstruction, pc);
+            Err(e) => {
+                tracing::trace!("DecodeError at {pc:X}: {e:?}");
+                let code = match e {
```

Comment:
It's probably worth implementing the `From` trait (e.g.: https://github.com/icicle-emu/icicle-emu/blob/2706b71b0275b0ceae2aa0ee9b83aef1866213b4/icicle-cpu/src/lib.rs#L312) to match how the rest of individual subsystem errors are mapped.